### PR TITLE
refactor(site): verify external auth before display ws form

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
image

### CODE REVIEW

File: site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import { ExternalAuthPollingState } from "../CreateWorkspacePage";
+import { ExternalAuthItem } from "./ExternalAuthItem";
+
+type ExternalAuthBannerProps = {
+  providers: TemplateVersionExternalAuth[];
+  pollingState: ExternalAuthPollingState;
+  onStartPolling: () => void;
+};
+
+export const ExternalAuthBanner = ({
```
Comment (aslilac) [inline site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthBanner.tsx:12]: `FC<ExternalAuthBannerProps>` pls

File: site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthBanner.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  id: "",
+  type: "",
+  display_name: "GitHub",
+  display_icon: "/icon/github.svg",
+  authenticate_url: "",
+  authenticated: false,
+};
+
+const meta: Meta<typeof ExternalAuthBanner> = {
+  title: "pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthBanner",
```
Comment (aslilac) [inline site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthBanner.stories.tsx:15]: I don't think adding the extra folder to storybook adds any benefit personally, and would suggest just naming it `"pages/CreateWorkspacePage/ExternalAuthBanner"`

File: site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthItem.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import Box from "@mui/material/Box";
+
+type Status = "idle" | "connecting";
+
+type ExternalAuthItemProps = {
+  provider: TemplateVersionExternalAuth;
+  isPolling: boolean;
+  defaultStatus?: Status;
+  onStartPolling: () => void;
+} & ComponentProps<typeof Box>;
```
Comment (aslilac) [inline site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthItem.tsx:17]: can we just import `BoxProps` instead?

File: site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthItem.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+export const ExternalAuthItem = ({
```
Comment (aslilac) [inline site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthItem.tsx:19]: `FC<ExternalAuthItemProps>` pls

**Review by aslilac — APPROVED** [review]
👍

File: site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthItem.tsx (hunk 4 above)
Comment (BrunoQuaresma) [reply site/src/pages/CreateWorkspacePage/ExternalAuthBanner/ExternalAuthItem.tsx:19]: I have to get used to it, sorry

**Comment by BrunoQuaresma** [comment]
Related to https://github.com/coder/coder/issues/11742

**Pull request merged**